integration-tests/oidc-wiremock/src/test/java/io/quarkus/it/keycloak/BearerTokenAuthorizationTest.java

```diff
@@ -0,0 +1,172 @@
+                    .statusCode(200)
+                    .body("userName", equalTo(username));
+        }
+    }
+
+    @Test
+    public void testAccessAdminResource() {
+        RestAssured.given().auth().oauth2(getAccessToken("admin"))
+                .when().get("/api/admin")
+                .then()
+                .statusCode(200)
+                .body(Matchers.containsString("granted:admin"));
+    }
+
+    @Test
+    public void testAccessAdminResourceCustomHeaderNoBearerScheme() {
+        RestAssured.given().header("X-Forwarded-Authorization", getAccessToken("admin"))
+                .when().get("/api/admin")
+                .then()
+                .statusCode(401);
+    }
+
+    @Test
+    public void testAccessAdminResourceCustomHeaderBearerScheme() {
+        RestAssured.given().header("X-Forwarded-Authorization", getAccessToken("admin"))
+                .when().get("/api/admin")
+                .then()
+                .statusCode(401);
+    }
+
+    @Test
+    public void testAccessAdminResourceWithRefreshToken() {
+        RestAssured.given().auth().oauth2(getRefreshToken("admin"))
+                .when().get("/api/admin")
+                .then()
+                .statusCode(401);
+    }
+
+    @Test
+    public void testPermissionHttpInformationProvider() {
+        RestAssured.given().auth().oauth2(getAccessToken("alice"))
+                .when().get("/api/permission/http-cip")
+                .then()
+                .statusCode(200)
+                .body("preferred_username", equalTo("alice"));
+    }
+
+    @Test
+    public void testDeniedAccessAdminResource() {
+        RestAssured.given().auth().oauth2(getAccessToken("alice"))
+                .when().get("/api/admin")
+                .then()
+                .statusCode(403);
+    }
+
+    @Test
+    public void testDeniedNoBearerToken() {
+        RestAssured.given()
+                .when().get("/api/users/me").then()
+                .statusCode(401);
+    }
+
+    //see https://github.com/quarkusio/quarkus/issues/5809
+    @RepeatedTest(20)
+    public void testOidcAndVertxHandler() {
+        RestAssured.given().auth().oauth2(getAccessToken("alice"))
+                .when().body("Hello World").post("/vertx")
+                .then()
+                .statusCode(200)
+                .body(equalTo("Hello World"));
+    }
+
+    @Test
+    public void testExpiredBearerToken() throws InterruptedException {
+        String token = getAccessToken("alice");
+
+        await()
+                .pollDelay(3, TimeUnit.SECONDS)
+                .atMost(5, TimeUnit.SECONDS).until(
+                        () -> RestAssured.given().auth().oauth2(token).when()
+                                .get("/api/users/me").thenReturn().statusCode() == 401);
+    }
+
+    private String getAccessToken(String userName) {
+        return Jwt.claim("preferred_username", userName)
+                .groups(new HashSet<>(Arrays.asList("user", "admin")))
+                .sign();
```

Comment:
At the moment the builder won't pick up the jwk kid itself, so please add, just in front of `.sign()`: `.jws.keyId("1")` (=> it has to match the kid value you will set in the public JWK key)